staging/src/k8s.io/apiserver/pkg/endpoints/handlers/patch.go

```diff
@@ -92,16 +92,21 @@ func PatchResource(r rest.Patcher, scope RequestScope, admit admission.Interface
 			scope.Serializer.DecoderToVersion(s.Serializer, schema.GroupVersion{Group: gv.Group, Version: runtime.APIVersionInternal}),
 		)
 
-		updateAdmit := func(updatedObject runtime.Object, currentObject runtime.Object) error {
-			if admit != nil && admit.Handles(admission.Update) {
-				userInfo, _ := request.UserFrom(ctx)
-				return admit.Admit(admission.NewAttributesRecord(updatedObject, currentObject, scope.Kind, namespace, name, scope.Resource, scope.Subresource, admission.Update, userInfo))
+		userInfo, _ := request.UserFrom(ctx)
+		updateMutateAdmit := func(updatedObject runtime.Object, currentObject runtime.Object) error {
```

Comment:
These functions seem generally useful? Why declare them as closures instead of putting them where the interfaces are declared?

Reply:
we only have them once. IMO they look like canonical cases of closures: fixing some arguments from the context and returning another functions with arguments.